gazebo_plugins/src/gazebo_ros_block_laser.cpp

```diff
@@ -304,16 +304,20 @@ void GazeboRosBlockLaser::PutLaserData(common::Time &_updateTime)
       j3 = hja + vjb * rayCount;
       j4 = hjb + vjb * rayCount;
       // range readings of 4 corners
-      r1 = std::min(this->parent_ray_sensor_->LaserShape()->GetRange(j1) , maxRange-minRange);
-      r2 = std::min(this->parent_ray_sensor_->LaserShape()->GetRange(j2) , maxRange-minRange);
-      r3 = std::min(this->parent_ray_sensor_->LaserShape()->GetRange(j3) , maxRange-minRange);
-      r4 = std::min(this->parent_ray_sensor_->LaserShape()->GetRange(j4) , maxRange-minRange);
+      r1 = this->parent_ray_sensor_->LaserShape()->GetRange(j1);
+      r2 = this->parent_ray_sensor_->LaserShape()->GetRange(j2);
+      r3 = this->parent_ray_sensor_->LaserShape()->GetRange(j3);
+      r4 = this->parent_ray_sensor_->LaserShape()->GetRange(j4);
 
       // Range is linear interpolation if values are close,
       // and min if they are very different
       r = (1-vb)*((1 - hb) * r1 + hb * r2)
          +   vb *((1 - hb) * r3 + hb * r4);
 
+      // clamp r to the min and max range of the sensor
+      r = std::min(r, maxRange);
+      r = std::max(r, minRange);
```

Comment:
> The range value was being clipped to (-infinity, maxRange-minRange], but it should be clipped to [minRange, maxRange]

Agreed that the previous clipping was weird. But accroding to [REP 117](https://www.ros.org/reps/rep-0117.html), shouldn't the out of range values become +-inf?

Reply:
Good catch. I changed to +- inf in ad7edd2 and adjusted the test world so their should be both -inf and inf values, but it turns out it doesn't matter. The gazebo sensor only casts rays from min to max range anyways because that's taken from the `sdf` xml tags, so values outside [minRange, maxRange] are impossible.

I don't think I can make time to come up for a fix for the block laser plugin never reporting values outside of min/max range, but this PR is still an improvement since it fixes the values being `minRange` too big.